Sources/SwiftDocC/Utility/MarkupExtensions/ListItemExtractor.swift

```diff
@@ -613,91 +126,97 @@ struct TaggedListItemExtractor: MarkupRewriter {
+        guard let extractedTag = listItem.extractTag() else {
+            return listItem
+        }
+        
+        switch extractedTag.knownTag {
+        case .returns:
+            // - Returns: ...
+            returns.append(.init(extractedTag))
+            
+        case .throws:
+            // "Throws" asides are currently (still) parsed as blockquote-style asides
+            return listItem
+            
+        case .parameter(let name):
+            // - Parameter x: ...
+            parameters.append(.init(extractedTag, name: name, isStandalone: true))
+            
+        case .parameters:
             // - Parameters:
             //   - x: ...
             //   - y: ...
-            parameters.append(contentsOf: parameterDescriptions)
-            return nil
-        } else if let parameterDescription = listItem.extractStandaloneParameter() {
-            // - parameter x: ...
-            parameters.append(parameterDescription)
-            return nil
-        } else if let dictionaryKeyDescription = listItem.extractDictionaryKeyOutline() {
+            parameters.append(contentsOf: listItem.extractInnerTagOutline().map { .init($0, name: $0.rawTag, isStandalone: false) })
+            
+        case .dictionaryKey(let name):
+            // - DictionaryKey x: ...
+            dictionaryKeys.append(.init(extractedTag, name: name))
+            
+        case .dictionaryKeys:
             // - DictionaryKeys:
             //   - x: ...
             //   - y: ...
-            dictionaryKeys.append(contentsOf: dictionaryKeyDescription)
-            return nil
-        } else if let dictionaryKeyDescription = listItem.extractStandaloneDictionaryKey() {
-            // - dictionaryKey x: ...
-            dictionaryKeys.append(dictionaryKeyDescription)
-            return nil
-        } else if let httpParameterDescription = listItem.extractHTTPParameterOutline() {
-            // - HTTPParameters:
+            dictionaryKeys.append(contentsOf: listItem.extractInnerTagOutline().map { .init($0, name: $0.rawTag) })
+        
+        case .httpResponse(let name):
+            // - HTTPResponse x: ...
+            httpResponses.append(.init(extractedTag, name: name))
+            
+        case .httpResponses:
+            // - HTTPResponses:
             //   - x: ...
             //   - y: ...
-            httpParameters.append(contentsOf: httpParameterDescription)
-            return nil
-        } else if let httpParameterDescription = listItem.extractStandaloneHTTPParameter() {
-            // - HTTPParameter x: ...
-            httpParameters.append(httpParameterDescription)
-            return nil
-        } else if let httpBodyDescription = listItem.extractHTTPBody() {
-            // - httpBody: ...
+            httpResponses.append(contentsOf: listItem.extractInnerTagOutline().map { .init($0, name: $0.rawTag) })
+            
+        case .httpBody:
+            // - HTTPBody: ...
             if httpBody == nil {
```

Comment:
Can you use `if let httpBody` here and avoid the `httpBody?.contents` syntax in the else clause?

Reply:
If so, there are a couple more places these appear ([1](https://github.com/swiftlang/swift-docc/pull/1014/files#diff-16e0f749ff7a0ce4a219f84e2331801278ed97e02b8e39ed9577deeb73b7b0fdR204), [2](https://github.com/swiftlang/swift-docc/pull/1014/files#diff-16e0f749ff7a0ce4a219f84e2331801278ed97e02b8e39ed9577deeb73b7b0fdR204))

Reply:
We could use `if let` here but we would need to assign back to `self.httpBody` since a local unwrapped copy would shadow the `httpBody` name within the scope of that if statement. 

In other words, the code would look something like:

```swift
if var httpBody {
    httpBody.contents = extractedTag.contents
    self.httpBody = httpBody
} else {
    httpBody = HTTPBody(mediaType: nil, contents: extractedTag.contents)
}
```

Both styles are fine. My main reason for using the `httpBody?.contents =` style is that the previous code used that style. However, I also believe that there's a slight risk with the `if var` style that code forgets to assign back to `self.httpBody` and thus accidentally drops some content.